connect/runtime/src/test/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerderTest.java

```diff
@@ -1168,25 +1090,27 @@ private static Map<String, String> taskConfig(SourceSink sourceSink) {
     }
 
     private void expectConfigValidation(
-            Connector connectorMock,
-            boolean shouldCreateConnector,
+            SourceSink sourceSink,
             Map<String, String>... configs
     ) {
         // config validation
+        Connector connectorMock = sourceSink == SourceSink.SOURCE ? mock(SourceConnector.class) : mock(SinkConnector.class);
         when(worker.configTransformer()).thenReturn(transformer);
         final ArgumentCaptor<Map<String, String>> configCapture = ArgumentCaptor.forClass(Map.class);
         when(transformer.transform(configCapture.capture())).thenAnswer(invocation -> configCapture.getValue());
         when(worker.getPlugins()).thenReturn(plugins);
         when(plugins.connectorLoader(anyString())).thenReturn(pluginLoader);
         when(plugins.withClassLoader(pluginLoader)).thenReturn(loaderSwap);
-        if (shouldCreateConnector) {
-            when(worker.getPlugins()).thenReturn(plugins);
-            when(plugins.newConnector(anyString())).thenReturn(connectorMock);
-        }
+
+        // Assume the connector should always be created
+        when(worker.getPlugins()).thenReturn(plugins);
+        when(plugins.newConnector(anyString())).thenReturn(connectorMock);
         when(connectorMock.config()).thenReturn(new ConfigDef());
 
-        for (Map<String, String> config : configs)
+        // Set up validation for each config
+        for (Map<String, String> config : configs) {
             when(connectorMock.validate(config)).thenReturn(new Config(Collections.emptyList()));
+        }
     }
 
     // We need to use a real class here due to some issue with mocking java.lang.Class
```

Comment:
nit: My IDE is also warning me that these should be `static`, which is fine for how these are used. Could you also fix that?